src/libraries/System.Text.Json/gen/JsonSourceGenerator.Emitter.cs

```diff
@@ -1370,6 +1392,70 @@ string GetParamDefaultValueAsString(object? value, Type type, string typeRef)
 
                 string FormatNumber() => $"({typeRef})({Convert.ToString(value, CultureInfo.InvariantCulture)})";
             }
+
+            private static readonly Dictionary<string, string> AliasedTypeMap = new Dictionary<string, string>()
+            {
+                [nameof(Boolean)] = "bool",
+                [nameof(Byte)] = "byte",
+                [nameof(Char)] = "char",
+                [nameof(Decimal)] = "decimal",
+                [nameof(Double)] = "double",
+                [nameof(Single)] = "float",
+                [nameof(Int32)] = "int",
+                [nameof(Int64)] = "long",
+                [nameof(Object)] = "object",
+                [nameof(SByte)] = "sbyte",
+                [nameof(Int16)] = "short",
+                [nameof(String)] = "string",
+                [nameof(UInt32)] = "uint",
+                [nameof(UInt64)] = "ulong",
+                [nameof(UInt16)] = "ushort"
+            };
+
+            private static string FormatTypeForXmlComment(Type type, string typeRef)
+            {
+                string GetFriendlyName(string typeName)
+                {
+                    if (AliasedTypeMap.TryGetValue(typeName, out string aliasedName))
+                    {
+                        return aliasedName;
+                    }
+                    return typeName;
```

Comment:
Even though the C# compiler does not allow this, it is possible to have IL type names to contain XML escape characters:

https://sharplab.io/#v2:DYLgZgzgNALiCWwA+MCeAHApgAgAa4D4CBJAcgFtsBDbANyuHgBNs0tsA7K8zAHl/zYAvNgDGwKhAjZMHJkA

Bare minimum, there needs to be some sort of detection for such pathological type names. 

That being said, we're fairly late in the development cycle for .NET 7, injecting arbitrary strings into XML docs makes me nervous that we might introduce unintended build-time regressions to the source generator. I don't see huge value in echoing the type name in the comments anyway, so I would recommend simply emitting constant XML comments for now.

Reply:
This makes a lot of sense to me in hindsight - it will also mean that there will be consistency between the comments. I hadn't considered the fact that potentially large types could cause the source generator (which is already quite complex) to noticeably slow down a build.